### PR TITLE
Reworks DiM and LM Robust variance estimators

### PR DESCRIPTION
**Pull request by lukesonnet**
This PR does quite a lot---more than my usual PRs.

**Difference in means:** this PR reworks some of the DoF and variance choices for the difference in means package. The specifications can be seen in the technical appendix, which can be most easily viewed right now here: http://htmlpreview.github.io/?https://github.com/DeclareDesign/estimatr/blob/lukesonnet/dim_dof/docs/articles/technical-notes.html

**lm_robust:** this PR also fixes a bug with the finite sample correction in HC1 standard errors, adds the option to get "stata" robust SEs (which is just HC1), and adds CR0 standard errors for the clustered case.

**Documentation:** reworked technical notes (linked above) that are up to date and close to ready for us to go on CRAN.

**Prepping for CRAN:** version changes, making it require randomizr 0.8, and a few other small changes.

### CODE REVIEW

**Comment by lukesonnet** [comment]
Yes, no problem. Will fix.

File: DESCRIPTION [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
              person("Luke", "Sonnet", email = "luke.sonnet@gmail.com", role = c("ctb")),
-             person("Neal", "Fultz", email="nfultz@gmail.com", role = c("ctb")))
-Description: This package provides fast estimators for common models in social science research.
+             person("Neal", "Fultz", email = "nfultz@gmail.com", role = c("ctb")))
+Description: Quickly estimate common models in social science research with robust standard errors.
 URL: http://estimatr.declaredesign.org, https://github.com/DeclareDesign/estimatr
 BugReports: https://github.com/DeclareDesign/estimatr/issues
 License: MIT + file LICENSE
-Depends:	R (>= 2.10)
+Depends: R (>= 2.10)
```
Comment (nfultz) [inline DESCRIPTION]: Might as well bump this to at least 3.3, that's our oldest supported version for this release cycle.

**Comment by lukesonnet** [comment]
Very sorry about the pkgdown update. Makes sense.

**Comment by lukesonnet** [comment]
Most of these doc changes have to do with the changes in the actual estimators so they fit with this PR, but I'm going to merge this in to master now so future changes go elsewhere.

**Pull request merged**